chain-signatures/node/src/protocol/message.rs

```diff
@@ -964,70 +937,33 @@ impl MessageOutbox {
                     .inc_by(partition.messages.len() as f64);
 
                 let client = self.client.clone();
-                send_tasks.push(tokio::spawn(async move {
-                    let start = Instant::now();
-                    if let Err(err) = client.msg(url, &[&encrypted_partition]).await {
-                        crate::metrics::NUM_SEND_ENCRYPTED_FAILURE
-                            .with_label_values(&[account_id.as_str()])
-                            .inc_by(partition.messages.len() as f64);
-                        crate::metrics::FAILED_SEND_ENCRYPTED_LATENCY
-                            .with_label_values(&[account_id.as_str()])
-                            .observe(start.elapsed().as_millis() as f64);
-                        Err(((from, to), partition, err))
-                    } else {
-                        crate::metrics::SEND_ENCRYPTED_LATENCY
-                            .with_label_values(&[account_id.as_str()])
-                            .observe(start.elapsed().as_millis() as f64);
-                        Ok(partition.messages.len())
-                    }
-                }));
-            }
-        }
-
-        let mut errors = Vec::new();
-        let mut retry = VecDeque::new();
-        let mut uncompacted = 0;
-        let mut compacted = 0;
-        for task in send_tasks {
-            match task.await {
-                Ok(Ok(msgs_len)) => {
-                    uncompacted += msgs_len;
-                    compacted += 1;
-                }
-                Ok(Err((route, partition, err))) => {
-                    // since we failed, put back all the messages related to this
-                    retry.push_back((route, partition));
-                    errors.push(err);
-                }
-                Err(err) => {
-                    tracing::warn!(?err, "outbox: task failed to send message");
-                }
+                tokio::spawn(timeout(
+                    Duration::from_millis(cfg.message_timeout),
```

Comment:
Optional suggestion: Consider logging and/or collecting metrics for messages we drop afer the tokio task gets cancelled after `Duration::from_millis(cfg.message_timeout)`. Something equivalent to the previous `tracing::warn!(?timeouts, "messages expired");` in the deleted `fn expire()`.

At the moment, it seems we only log warnings on each retry iteration but we don't see anywhere if it eventually succeeded or if it failed in the end.